test/e2e-go/features/transactions/messageRate_test.go

```diff
@@ -0,0 +1,264 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package transactions
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/account"
+	"github.com/algorand/go-algorand/libgoal"
+	"github.com/algorand/go-algorand/test/framework/fixtures"
+	"github.com/algorand/go-algorand/util/db"
+)
+
+// TODO: remove print statements
+
+// this test checks that the txsync outgoing message rate
+// varies according to the transaction rate
+func TestMessageRateChangesWithTxnRate(t *testing.T) {
+	txnRates := []uint{50, 300, 800, 1300}
+	if testing.Short() {
+		txnRates = []uint{50, 300}
+	}
+	testMessageRateChangesWithTxnRate(t, filepath.Join("nettemplates", "OneNodeTwoRelays.json"), txnRates)
+}
+
+func throttleTransactionRate(startTime time.Time, txnRate uint, sentSoFar uint) {
+	timeDelta := time.Since(startTime)
+	currentTps := float64(sentSoFar) / timeDelta.Seconds()
+	if currentTps > float64(txnRate) {
```

Comment:
The test should also check here if the expected transaction rate is being achieved, and fail here if the rate is not possible on the system where the test is running. 
Otherwise, the test will fail blaming the message rate instead of the transaction rate.

Reply:
Fixed.